ONSdigital/takeon-business-layer
# Td/spp3283/improve linting

# Description
The purpose of this story is to rectify the linting errors in the existing code of Business layer Java files.  To keep the consistency, linting should be improved and then linting maintained going forward. The linting rules are maintained by Checkstyle plugin and the rules are easily configurable by modifying **checkstyle.xml**
**/takeon-business-layer/checkstyle.xml**

# How to test?
Compare the linting errors in the existing 'main' repository with the existing branch 'td/spp3283/improve-linting'. The tester would make sure there are no linting errors as those are rectified in the existing branch code.

Note: In the existing main repository there are several linting errors.

This can be done by running 
**mvn package install** in both 'main' and 'td/spp3283/improve-linting' branches

# Unit Tests
Make sure there are no tests are failing and this can be done by running java unit tests using: 

**mvn clean test**

# Code Coverage
You can verify code coverage using: awk -F"," '{ instructions += $4 + $5; covered += $5 } END { print covered, "/", instructions, " instructions covered"; print 100*covered/instructions, "% covered" }' target/site/jacoco/jacoco.csv

# Links
https://collaborate2.ons.gov.uk/jira/browse/SPP-3283
